go/database/mpt/verification.go

```diff
@@ -41,15 +42,63 @@ func (NilVerificationObserver) StartVerification()        {}
 func (NilVerificationObserver) Progress(msg string)       {}
 func (NilVerificationObserver) EndVerification(res error) {}
 
-// VerifyFileForest runs list of validation checks on the forest stored in the given
+// VerifyMptState runs validation checks on the forest and code hashes
+// stored in the given directory.
+// Forest checks:
+//   - all required files are present and can be read
+//   - all referenced nodes are present
+//   - all hashes are consistent
+//
+// Code Hashes checks:
+//  1. Fatal checks
+//     - all CodeHashes within the Mpt are present in the code file
+//     - all CodeHashes within the code file are correct
+//  2. Non-fatal checks
+//     - there are no extra Code Hashes not referenced by any account
+func VerifyMptState(directory string, config MptConfig, roots []Root, observer VerificationObserver) (res error) {
+	if observer == nil {
+		observer = NilVerificationObserver{}
+	}
+
+	observer.StartVerification()
+	defer func() {
+		if r := recover(); r != nil {
+			panic(r)
+		}
+		observer.EndVerification(res)
+	}()
+
+	// Open stock data structures for content verification.
+	observer.Progress("Obtaining read access to files ...")
+	source, err := openVerificationNodeSource(directory, config)
+	if err != nil {
+		return err
+	}
+	defer source.Close()
+
+	err = verifyContractCodes(directory, source, observer)
+	if err != nil {
+		return err
+	}
+
+	err = verifyForest(directory, config, roots, source, observer)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// verifyFileForest runs list of validation checks on the forest stored in the given
 // directory. These checks include:
 //   - all required files are present and can be read
 //   - all referenced nodes are present
 //   - all hashes are consistent
-func VerifyFileForest(directory string, config MptConfig, roots []Root, observer VerificationObserver) (res error) {
+func verifyFileForest(directory string, config MptConfig, roots []Root, observer VerificationObserver) (res error) {
```

Comment:
what is actually reason to change this method to private? It seems unrelated to work of this PR.

Reply:
well, there was no reason even before, because it was never called outside of this package. We can talk if it should have been private already, but I prefer if PRs really change only what is related, and potentially reduction of technical debt is done in a separate PR.